bugsnag-android-core/src/main/java/com/bugsnag/android/Configuration.java

```diff
@@ -0,0 +1,833 @@
+     */
+    public void setMaxBreadcrumbs(int maxBreadcrumbs) {
+        impl.setMaxBreadcrumbs(maxBreadcrumbs);
+    }
+
+    /**
+     * Bugsnag uses the concept of "contexts" to help display and group your errors. Contexts
+     * represent what was happening in your application at the time an error occurs.
+     *
+     * In an android app the "context" is automatically set as the foreground Activity.
+     * If you would like to set this value manually, you should alter this property.
+     */
+    @Nullable
+    public String getContext() {
+        return impl.getContext();
+    }
+
+    /**
+     * Bugsnag uses the concept of "contexts" to help display and group your errors. Contexts
+     * represent what was happening in your application at the time an error occurs.
+     *
+     * In an android app the "context" is automatically set as the foreground Activity.
+     * If you would like to set this value manually, you should alter this property.
+     */
+    public void setContext(@Nullable String context) {
+        impl.setContext(context);
+    }
+
+    /**
+     * Sets which values should be removed from any Metadata objects before
+     * sending them to Bugsnag. Use this if you want to ensure you don't send
+     * sensitive data such as passwords, and credit card numbers to our
+     * servers. Any keys which contain these strings will be filtered.
+     *
+     * By default, redactedKeys is set to "password"
+     */
+    @Nullable
+    public Set<String> getRedactedKeys() {
+        return impl.getRedactedKeys();
+    }
+
+    /**
+     * Sets which values should be removed from any Metadata objects before
+     * sending them to Bugsnag. Use this if you want to ensure you don't send
+     * sensitive data such as passwords, and credit card numbers to our
+     * servers. Any keys which contain these strings will be filtered.
+     *
+     * By default, redactedKeys is set to "password"
+     */
+    public void setRedactedKeys(@Nullable Set<String> redactedKeys) {
+        if (redactedKeys != null) {
+            impl.setRedactedKeys(redactedKeys);
+        } else {
+            impl.setRedactedKeys(Collections.<String>emptySet());
+        }
+    }
+
+    /**
+     * Allows you to specify the fully-qualified name of error classes that will be discarded
+     * before being sent to Bugsnag if they are detected. The notifier performs an exact
+     * match against the canonical class name.
+     */
+    @Nullable
+    public Set<String> getDiscardClasses() {
+        return impl.getDiscardClasses();
+    }
+
+    /**
+     * Allows you to specify the fully-qualified name of error classes that will be discarded
+     * before being sent to Bugsnag if they are detected. The notifier performs an exact
+     * match against the canonical class name.
+     */
+    public void setDiscardClasses(@Nullable Set<String> discardClasses) {
+        if (discardClasses != null) {
```

Comment:
Do we also need to check each String within the Set for null?

Reply:
Yep - we need to do this for all our collections. Should the behaviour be to not set the field in this case, or to carry on without that null value?

Reply:
From in-person discussion we decided to alter this so that if _any_ value in a collection is null, the whole parameter is treated as null